### ISSUE TITLE
Extracting ucd/desc/unit from FIELD/PARAM

### ISSUE THREAD

**Issue by lmichel** [ivoa-std/ModelInstanceInVot#5]
The role of MANGO parameters is given by both UCD and text Description. These values can be duplicated :

```xml
<ATTRIBUTE dmrole="mango:Parameter.ucd" dmtype="string" value="pos.eq"/>
<ATTRIBUTE dmrole="mango:Parameter.desc" dmtype="string" value="This is the position"/>
```
But it would be nice whether one could just pick these values from the FIELD/PARAM elements.
The same problem occurs wit the unit that have to be set  in most of the Coordinates

**Option #1** 
Adding new attributes to ATTRIBUTEs
```xml
<ATTRIBUTE dmrole="mango:Parameter.ucd" dmtype="string" ref_ucd="_field_id" value="pos.eq"/>
<ATTRIBUTE dmrole="mango:Parameter.desc" dmtype="string" ref_desc="_field_id"  value="This is the position"/><ATTRIBUTE dmrole="coord:Val.unit" dmtype="string" ref_unit="_field_id"  value="deg"/>
```
This has the advantage of working with both PARAM and FIELD with the same rule as for the regular values

**Option #2** 
Creating new elements

```xml
<UCD dmrole="mango:Parameter.ucd" dmtype="string" ref="_field_id" value="pos.eq"/>
<DESC dmrole="mango:Parameter.desc" dmtype="string" ref="_field_id"  value="This is the position"/>
<UNIT dmrole="coord:Val.unit" dmtype="string" ref="_field_id"  value="deg"/>
```
For my perspective, this is just annoying for the parsers (less elements we have better is is)

**Comment by lmichel** [comment]
Can we consolidate Option#1?

**Comment by mcdittmar** [comment]
```xml
<ATTRIBUTE dmrole="mango:Parameter.ucd" dmtype="string" value="pos.eq"/>
<ATTRIBUTE dmrole="mango:Parameter.desc" dmtype="string" value="This is the position"/>
<ATTRIBUTE dmrole="coord:Val.unit" dmtype="string" value="deg"/>
```

Here is my view on these:
**UNIT:**
* An ivoa:Quantity type is fully defined by a single reference to a VOTable PARAM/FIELD element.
  There is no need for individually annotating the Quantity.value and Quantity.unit unless the serialization stores these in different VOTable elements ( ie: different FIELDs/PARAMs)
   For example, (using the same syntax as above)..
```xml
<!-- when unit is included in PARAM/FIELD -->
<ATTRIBUTE dmrole="coords:PhysicalCoordinate.cval" dmtype="ivoa:RealQuantity" ref="_mag"/>

<!-- when unit and value individually serialized -->
<INSTANCE dmtype="ivoa:RealQuantity">
  <ATTRIBUTE dmrole="coords:PhysicalCoordinate.cval.value" dmtype="ivoa:real" ref="_mag"/>
  <ATTRIBUTE dmrole="coords:PhysicalCoordinate.cval.unit" dmtype="ivoa:Unit" ref="_mag_unit"/>
</INSTANCE>
```
* As a LITERAL, there may be an open issue.
    * VM syntax has a LITERAL with value and unit attributes, so can fully define an ivoa:Quantity type.
    * MIVOT syntax ATTRIBUTE element does not include unit, so one cannot define a literal using (dmtype, value, unit ) attributes.
* The merged syntax element serving literals should include a 'unit' attribute.
```xml
<ATTRIBUTE dmrole="coords:PhysicalCoordinate.cval" dmtype="ivoa:RealQuantity" value="6246.77" unit="Angstrom" />
```

**UCD:**
* The usage we are discussing for this ( to identify the 'physics' of a GenericMeasure ) re-uses the ucd concept, but the actual value is usually(?) not the same as what is associated with any individual FIELD/PARAM.  In the workshop examples we are often seeing multi-word qualified values such as  ucd="pos.eq" or "pos.eq.ra" or "phot.mag;em.opt", where in this usage, I'd expect to see the simplest top-level word "pos" or "phot" or "time" . 
* So, I'm not seeing a pressing need to pull these from the PARAM/FIELD content.  Forcing separate declaration of the value following the model requirements seems the proper approach.

**DESCRIPTION:**
* This is proving harder to write than I thought.. I'll come back to this after lunch.

**Comment by mcdittmar** [comment]
**DESCRIPTION:**
* My understanding about this is that it should be similar to "ID" in models.. all objects MAY have a description, but it is not a modeled feature.  Instead, IF your application cares to convey this information, it can pull it from the annotated votable element and added to the instance.
* However, this has come up in Mango and Provenance.. so lets see.
    * In the example above, the Parameter object has an explicit 'desc' attribute 
        * this implies that it is a significant element in the understanding of the Parameter object, which I'm not sure is true here.
        * in the Provenance usage, the 'description' attributes were more integral to the Description classes.. like a form with a description field.
    * as such, the value must be supported by both reference and in literals.
        * pulled from the value of a FIELD/PARAM (normal ```@ref``` access)
        * provided as a literal string by the provider/annotator (normal ```@value``` access)
    * and this request would be for the value to be
        * pulled off the DESCRIPTION sub-element of a VOTable FIELD/PARAM. (new)
    * I assume that this access mode is not useful for anything but an Object.description like attribute.
      We aren't expecting this to be used to extract anything but descriptive text... like "photdm-alt:MagnitudeSystem.type"
	* NOTE: DESCRIPTION is also allowed on VOTable elements VOTABLE, RESOURCE, TABLE, GROUP
	  so, in theory, one could pull the description of an INSTANCE from a GROUP, or a SparseCube from the VOTABLE or RESOURCE.
* If the above is true, then I'd say this access mode fits into my original point above and we should
    * structure the models such that if the model attribute value is not expected to be pulled from a FIELD/PARAM value,
it should not be an attribute.
    * and for annotation either:
        1) do nothing in the annotation; and leave it to implementations to decide if they want to pull the VOTable DESCRIPTION content into the instances which use them.
        2) handle it similar to VOTable DESCRIPTION and define an optional DESCRIPTION sub-element contained in INSTANCE/ATTRIBUTE.  
            * Support a reference to the content via ID to the containing VOTable element, this ONLY pulls from the DESCRIPTION sub-element of the referenced VOTable element.
            * or as inline text (literal).  Since this can be quite long, I'm not sure if an attribute is the proper location for this.
```xml
<INSTANCE dmtype="mango:Parameter">
  <DESCRIPTION ref={votable_element_id}>
    { value provided as literal, which can be quite lengthy }
  </DESCRIPTION>
</INSTANCE>
```

**Comment by lmichel** [comment]
**General comment:** I'm totally in line with Markus when he urges no to duplicate VOTable meta-data in the mapping. If we do not duplicate them, we have to refer to them.

**Unit**

> The merged syntax element serving literals should include a 'unit' attribute.

You are just reinventing my SC_QUANTITY :-)

The problem is that we are talking about syntax, we cannot assume that models will use ivoa:Quantity (I really dislike this pattern but this is not the topic here). 
We need then a solution to set unit values. We have 2 options:
- set them as literal as shown above
- pick them from the VOTable. The problem is that units are not in table columns, thus your example with `ref=_mag_unit`  does not work. We need a specific mechanism telling that the value must be picked out of the FIELD. That is the purpose of my proposal. `unit_ref=_mag_unit`

**Comment by lmichel** [comment]
**UCD** 

UCDs have been set by the curator and we can assume they are fine. There is no reason to prevent reusing them in the annotations as well as there is non  reason to force to use them (e.g. your position example)
Thus the annotation must give the choice. 
This is the purpose of my proposal that works with both reference and literals.
```xml
<ATTRIBUTE dmrole="mango:Parameter.ucd" dmtype="string" ref_ucd="_field_id" value="pos.eq"/>
```

**Description** 

Can be tackle with the same way.


I do not see what is wrong with this proposal.

**Comment by lmichel** [comment]
> My understanding about this is that it should be similar to "ID" in models.. all objects MAY have a description, but it is not a modeled feature

Not agreed: We are not talking about models here we are talking about annotations. As you said, we have 2 models having free text descriptions as attributes, thus we have to deal with this. Considering that a model instances must be totally self-describing, this sort of feature makes sense by the way.

> structure the models such that if the model attribute value is not expected to be pulled from a FIELD/PARAM value,
it should not be an attribute.

I'll never follow you on this way. The annotation must support models that reuse VOTable meta-data.
We have three cases where we need to extract things from FIELD/PARAM attributes (not value).

**Comment by lmichel** [comment]
As you look anxious to use (MIVOT) ATTRIBUTE for something different than extracting FIELD/PARAM values, I try a proposition that could avoid the role mixing you are worry about:

**Option 4** 
Creating a new element for this purpose
```xml
<FIELD_ATT dmrole="mango:Parameter.ucd" dmtype="string" ref_ucd="_field_id" value="pos.eq"/>
<FIELD_ATT dmrole="mango:Parameter.desc" dmtype="string" ref_desc="_field_id"  value="This is the position"/>
<FIELD_ATT dmrole="coord:Val.unit" dmtype="string" ref_unit="_field_id"  value="deg"/>
```
This has the advantage of working with both PARAM and FIELD with the same rule as for regular values.
This needs a exclusion rule for all the `@ref` flavors.

**Comment by mcdittmar** [comment]
> **General comment:** I'm totally in line with Markus when he urges no to duplicate VOTable meta-data in the mapping. If we do not duplicate them, we have to refer to them.
> 
> **Unit**
> 
> > The merged syntax element serving literals should include a 'unit' attribute.
> 
> You are just reinventing my SC_QUANTITY :-)

No.. SC_QUANTITY was a separate element.  By "element serving literals" I mean the syntax element dealing with leaves.
I left it vague because we have not resolved that issue yet.  Specifically I mean:
VM: LITERAL( dmtype, value, unit ) - already has unit
MIVOT: ATTRIBUTE( dmrole, dmtype, ( ref | value ) ) - would add a 'unit'
MERGED currently:  INSTANCE( dmtype, ( ref | value ) ) _/

> 
> The problem is that we are talking about syntax, we cannot assume that models will use ivoa:Quantity (I really dislike this pattern but this is not the topic here).

We don't have a counter-example yet, and I do think it's safe to assume that alternatives to ivoa:Quantity will still combine/associate a value + unit combination, which is still equivalent to a PARAM and the same convention may apply to it.

> We need then a solution to set unit values. We have 2 options:
> * set them as literal as shown above
> * pick them from the VOTable. The problem is that units are not in table columns, thus your example with `ref=_mag_unit`  does not work. We need a specific mechanism telling that the value must be picked out of the FIELD. That is the purpose of my proposal. `unit_ref=_mag_unit`

You misinterpreted the 'ref=+mag_unit' example.  The cases show how the annotation would handle the 3 cases:
1) model object (like ivoa:Quantity) mapping to a single PARAM|FIELD, providing both value + unit
    1) Single annotation of ivoa:Quantity like object to the PARAM
2) model object (like ivoa:Quantity) where value and unit are provided in separate PARAM|FIELD elements.
    1) Annotate ivoa:Quantity like object content to the corresponding PARAM/FIELD element
3) model object (like ivoa:Quanitty) provided as literal
    1) Single annotation using (value, unit) attributes of the syntax

Is there a situation in the use case examples where we have an issue annotating the content?

**Comment by mcdittmar** [comment]
> I left it vague because we have not resolved that issue yet. Specifically I mean:
> VM: LITERAL( dmtype, value, unit ) - already has unit
> MIVOT: ATTRIBUTE( dmrole, dmtype, ( ref | value ) ) - would add a 'unit' to accommodate the 'as literal' thread.
> MERGED currently: INSTANCE( dmtype, ( ref | value ) ) _/

I should add that I believe we need to add the 'unit' attribute here to support the 'as literal' thread regardless of the decision w.r.t. enabling a 'unit_ref' access directly to the unit attribute of the PARAM.

**Comment by mcdittmar** [comment]
> As you look anxious to use (MIVOT) ATTRIBUTE for something different than extracting FIELD/PARAM values, I try a proposition that could avoid the role mixing you are worry about:

Its not that I have any issue using MIVOT::ATTRIBUTE to provide the functionality, it is just that when I evaluate the examples, I'm not sure that this is functionality we need to provide.

* unit: is covered with annotation of ivoa:Quantity, or similar object directly to PARAM
* ucd: I don't agree that the param.ucd is the same ucd as what is in the one model example we have.
    * ie: the 'ucd' value from the several columns which make up a Measure/Property, is not the 'ucd' of the Measure/Property.
* desc: I think of this as ancillary information unless it is integral to the model element.  If it is integral, I don't expect that it would be serialized in the VOT::DESCRIPTION node.

I just don't want to add unnecessary content to the syntax.

With these features included, the signature for the element becomes (using MIVOT::ATTRIBUTE base)
* ATTRIBUTE( dmrole, dmtype, ref, ref_unit, ref_ucd, ref_desc, value, unit )

When you look at it like this, it implies that 'ucd' and 'desc' are parts of the ATTRIBUTE, when they are really saying "The 'value' of this ATTRIBUTE is given by the 'ucd' (or unit or desc) sub-attribute of the referenced PARAM|FIELD.

IF that were to be supported, I'd maybe rather see a secondary attribute, so 'ref' is always the PARAM reference, fine-tuned by the secondary.
* ATTRIBUTE( dmrole, dmtype, ref, subref, value, unit ) where subref = "ucd|unit|desc".

When I ask myself, what pieces of the use case examples can't be done without this?  I think the only item is the Property.description attribute, which I think is a model level discussion as much as syntax.

**Comment by lmichel** [comment]
> No.. SC_QUANTITY was a separate element.

SC_QUANTITY was a sort of macro enclosing the mapping of an ivo:realQuantity (see [p 19](https://github.com/ivoa-std/ModelInstanceInVot/blob/master/doc/model-instance-in-vot.pdf)) and unit are set as literal value in there (... unit="erg/sec/cm2" ...)

>  We don't have a counter-example yet, 

In a sky position you have 2 parameters (ra/dec) with the same unit. So  if you use 2 ivo:Quantity, you will have to duplicate the units with a risk  of inconsistency or even worse negative comments from our colleagues
In a general way, the mapping is model-agnostic: it must support value/unit pairs represented with  ivo:realQuantity or not.

(out of topic: I dislike ivo:Quantity, because the way to access a value depends on whether it has a unit of no.)

> You misinterpreted the 'ref=+mag_unit' example. 

I do  not see how do you make distinction between units  set as literal values and units read from  FIELD/PARAM@unit
Do you propose to do some inference on the DM type (dmtype == ivo:Qantity then take the FIELD unit)?

If you map a Quantity, you will get something like this
```xml
<INSTANCE dmrole="coords:PhysicalCoordinate.cval" dmtype="ivoa:RealQuantity">
    <ATTRIBUTE dmrole="ivoa:RealQuantity.value" dmtype="ivoa:real" ref="col_id" />
    <ATTRIBUTE dmrole="ivoa:RealQuantity.unit" dmtype="ivoa:Unit" value="m/sec" />
</INSTANCE>
```
In this case the unit is set as a literal `m/sec`
If I want to take the FIELD unit, I need something like:

```xml
<INSTANCE dmrole="coords:PhysicalCoordinate.cval" dmtype="ivoa:RealQuantity">
    <ATTRIBUTE dmrole="ivoa:RealQuantity.value" dmtype="ivoa:real" ref="col_id" />
    <FIELD_ATT dmrole="ivoa:RealQuantity.unit" dmtype="ivoa:Unit" unit_ref="col_id" /> <!-- or whatever syntax -->
</INSTANCE>
```

**Comment by lmichel** [comment]
> Its not that I have any issue using 

- unit:  see me previous post
- UCD: I do not say _param.ucd is the same ucd as what is in the one model_ , I say that _param.ucd might be the same ucd as what is in the one model_. If I had to develop an on-the-fly Mango annotation , I would take the native UCDs for all fields except those of an exception list. Therefore, having a way to reference a FIELD units is a rather convenient tool although I can also make copies.
- DESC: same as for UCDs.

The purpose an MANGO is to increase the interoperability of native (meta)data. The interweaving between model leaves and VOtable meta-data is deeper than with e.g.  CUBE. 

> ATTRIBUTE( dmrole, dmtype, ref, subref, value, unit ) where subref = "ucd|unit|desc".

I like this proposal.
To conclude: I would be happy to keep this feature for MANGO, but if you really hate it, we can leave withdraw it.

**Comment by mcdittmar** [comment]
> > We don't have a counter-example yet,
> 
> In a sky position you have 2 parameters (ra/dec) with the same unit. So if you use 2 ivo:Quantity, you will have to duplicate the units with a risk of inconsistency or even worse negative comments from our colleagues
> In a general way, the mapping is model-agnostic: it must support value/unit pairs represented with ivo:realQuantity or not.

Can you provide the VOTable elements for this?  I don't follow the scenario.  Each votable param/field will have a unit which is pulled into the corresponding Quantity along with the value.

> I do not see how do you make distinction between units set as literal values and units read from FIELD/PARAM@unit
> Do you propose to do some inference on the DM type (dmtype == ivoa:Quantity then take the FIELD unit)?

Ah.. here it is!  I feel like this detail has been missed in this discussion.  
Yes.. the ivoa:Quantity is designed to easily map into a VOTable PARAM|FIELD, and the VM syntax 
From VO-DML model spec, pg 58: "A particular motivation for defining this concept in the base model is its use in the mapping specification, which allows mapping of these structured DataType-s directly to single FIELDs or PARAMs. Here the unit specified on those VOTable elements is assumed to be mapped to the unit on the Quantity"

So, for most usage, you need only this one line to populate the full Quantity value and unit.
You only need to do the above expansion IF you need to individually specify the content.

```
 <ATTRIBUTE dmrole="coords:PhysicalCoordinate.cval" dmtype="ivoa:RealQuantity ref="col_id" \>
```

I have been assuming that this convention would be carried into the merged syntax.  
There is a section (6.3.6) in the VM syntax document.. empty, but there.. which would have given the details.

If/when we get a model which does not use the ivoa:Quantity type, I agree that we would need a mechanism for annotating that.  That same mechanism could apply to the 'ucd' and 'desc' sub-content as well..

This feature is specifically directed towards MANGO, while the coords,meas,ds,cube models do not require it.
So, what I'd prefer to do is to tag this as a v1.1 feature to be worked with the MANGO model.  This would allow the other models to proceed with the initial version of the annotation syntax.  

I think we've teased this out enough to see that it could be an additive (backward compatible) update.  And we've put out some options which may need review/prototyping in action before settling on the final solution.

**Comment by lmichel** [comment]
> Can you provide the VOTable elements for this

If you have a 2D coordinate space (e.g. the unit sphere), both coordinates must have the same unit otherwise the metric is wrong e.g. you cannot compute distances with RA in deg and DEC in rad.
It turns out that VOTable fields are defined in a independent way (one for RA and one for DEC) and then each has its own unit. But this is a degeneration of the coordinate space.
Units might differ when positions are expressed in sexadecimal, but in this case, values are strings (see [proper motion](https://github.com/ivoa/dm-usecases/blob/main/usecases/standard_properties/raw_data/vizier_propermotion.xml) and giving a unit to string values is odd as well. 
In fact this pattern just tells the client how the conversion into a proper unit has to be done.
I don't know how you will deal with this with your new sky position classes, but in term of mathematical consistency, longitude and latitude should share the same unit. 
This is why I thing the ivo:Quantity is not appropriate for sky position (same with proper motions)

**Comment by lmichel** [comment]
> This feature is specifically directed towards MANGO, while the coords,meas,ds,cube models do not require it.

This is true, but I would be sure that MANGO won't be stalled on the MAPPING1.1 RFC: I will be retired in 5 years! 
I propose to put this feature aside but I'll continue to work on it, based on your proposal, and I'd like to keep the door open to put it in the final spec.
As you said, adding/removing it doesn't harm anything

**Comment by lmichel** [comment]
> So, for most usage, you need only this one line to populate the full Quantity value and unit.
You only need to do the above expansion IF you need to individually specify the content.

I confirm that this was exactly the purpose of my `SC_QUANTITY`!
Not being far from a contradiction, I preferred to have a specific element for this in order not to have ATTRIBUTE elements behaving in a different way according to theirs `dmtypes`:

```xml
<ATTRIBUTE dmrole="coords:PhysicalCoordinate.cval" dmtype="ivoa:RealQuantity" ref="col_id" \>
```
will generate something like this because : `dmtype="ivoa:RealQuantity"`
```json
{ "value": 1234.5,
   "unit": "m/sec"}
```
whereas, for a fields not modeled with QUANTITY (flags,names... and see my above post)
```xml
<ATTRIBUTE dmrole="coords:PhysicalCoordinate.cval" dmtype="ivoa:real ref="col_id" \>
```
will generate something like this:
```json
{ "value": 1234.5}
```

The parser has to check the type to get what it has to do, which is something a prefer to avoid.
IMO it makes sense to have specific elements to process specific model types.
What do you think?

**Comment by mcdittmar** [comment]
> > Can you provide the VOTable elements for this
> 
> If you have a 2D coordinate space (e.g. the unit sphere), both coordinates must have the same unit otherwise the metric is wrong e.g. you cannot compute distances with RA in deg and DEC in rad.

Yes.. the position components need to be in the same units to be useful, if they are serialized in PARAM|FIELD with different units, then it is the responsibility of the interpreter to unify them.  This doesn't seem like an annotation issue.

> This is why I thing the ivo:Quantity is not appropriate for sky position (same with proper motions)

That is a modeling discussion.  There are many right answers when modeling this content.  The Quantity element maps very well to how we serialize the data, so is a good fit.

**Comment by lmichel** [comment]
> That is a modeling discussion.

Sure.
I'm just saying that a model associating an unit to RA and another to DEC is a bit odd since both axis have the same unit by construction. In this case it is legitimate not to use Quantity, thus the mapping syntax must be able to map `cval` that are not Quantities.

**Comment by mcdittmar** [comment]
> Not being far from a contradiction, I preferred to have a specific element for this in order not to have ATTRIBUTE elements behaving in a different way according to theirs `dmtypes`:
> 

* The mapping syntax should be model agnostic, by adding "SC_QUANTITY" type elements to the syntax, this is not true.
    * We do not want to add other annotation tag whenever a model wants to define a Quantity alternative.
* The parser must interpret the dmtype in order to know what sort of instance to generate.

> The parser has to check the type to get what it has to do, which is something a prefer to avoid.
> IMO it makes sense to have specific elements to process specific model types.
> What do you think?

I see your point, but have a different interpretation.
* the MIVOT::ATTRIBUTE (above) says "this is a leaf instance"
    * the content of this leaf is populated by the ```@ref, @value, @unit``` attributes
    * note: the MIVOT spec does not include 'unit', but is being added per this discussion, and aligning with the VM syntax.
* regardless of what the ```@dmtype``` of the ATTRIBUTE is, ( ivoa:RealQuantity, ivoa:real, photDM:PhysicalQuantityDouble), the interpreter should be attempting to assign the value and 'unit attribute' of the instance from the provided content.  
    * either the element attributes directly, or from the underlying PARAM|FIELD attributes or a combination.
* this assignment will either succeed or fail depending on the nature of the instance being created.

This works well for the ```@unit``` attribute since it is so tightly bound to the value itself.
It does not work for the ```@description and @ucd`` portions of a PARAM|FIELD in the case we are considering:
*  the value of this instance.xyz attribute is provided by the ```description sub-element``` of the referenced PARAM.
*  the value of this instance.xyz attribute is provided by the ```@ucd attribute``` of the referenced PARAM.

**Comment by lmichel** [comment]
> I see your point, but have a different interpretation.

Tie game: with `SC_QUANTITY` the model dependency is explicit whereas in your case it is hidden, but it does exist in both cases.
I see you really prefer `ATTRIBUTE/dmtype` solution, so let's go with this.

> either the element attributes directly, or from the underlying PARAM|FIELD attributes or a combination.

My understanding is that the parser takes unit of the referenced PARAM|FIELD if it exists and @unit otherwise (same mechanism as for  ref/value)

But the issue remains to get UCD/UNIT/DESC out of the scope of `ivo:Quantities`.
I hope I convinced you that even being very faith to VODML, the case might occur.
How do you map this
```xml
<FIELD name="pm_ra" unit="mas/y" ... />
<FIELD name="pm_dec" unit="mas/y" ... />
```
on a model like that 
```
{ 
ra :real
dec: real
unit : string
}

You will get something like

```xml
<INSTANCE dmrole="pm" dmtype="ProperMotion">
    <ATTRIBUTE dmrole="pm.ra" dmtype="real" ref="pm_ra"/>
    <ATTRIBUTE dmrole="pm.dec" dmtype="real" ref="pm_dec"/>
    <??????????? dmrole="pm.unit" dmtype="string" ref="pm_dec" ???????????/>
</INSTANCE>
```

I keep focused on the UNIT but the issue is the same with UCD/DESC

**Comment by mcdittmar** [comment]
> But the issue remains to get UCD/UNIT/DESC out of the scope of `ivo:Quantities`.
> I hope I convinced you that even being very faith to VODML, the case might occur.
> How do you map this
> 
> ```
> <FIELD name="pm_ra" unit="mas/y" ... />
> <FIELD name="pm_dec" unit="mas/y" ... />
> ```
> 
> on a model like that
> 
> ```
> { 
> ra :real
> dec: real
> unit : string
> }
> ```

Yes, this is possible but currently hypothetical.
I'd hope that we'd encourage models to use the ivoa base, but it is not required.

The only model needing this feature at the moment is Mango, and I feel there may still a good bit of discussion about the model itself to be done.  This is why I suggest we link this annotation feature with the Mango model, and let them progress together.   In the mean time, the other models-in-progress can proceed.

**Comment by lmichel** [comment]
I'm not sure that a LonLatSkyPosition class like

```json
{ ra :{ value:real,
          unit:string},
  dec :{ value:real,
          unit:string}
          }
```
will go through the RFC without damage!

You are right, 
- This a feature that would help for MANGO
- MANGO a far from the PR
- This may confuse people about the scope of the mapping

But I insist a little bit because:
- I know the VO time-scale
- If the mapping is really used within the next years people will be reluctant to go into 1.1 version. 
- This feature is simple to implement
- No conflict with other elements
- It look wise to secure it while we have a good momentum

**Comment by mcdittmar** [comment]
> I'm not sure that a LonLatSkyPosition class like
> 
> ```
> { ra :{ value:real,
>           unit:string},
>   dec :{ value:real,
>           unit:string}
>           }
> }
> ```
> 
> will go through the RFC without damage!

OK.. but that is what you asked for.
This particular concern has not been raised in the past (I think), even with Markus' appeal to separate the models, it would use Quantity as the basis.  And, as a 3D point, the unit would need to be at the axis level.

If there is such a concern, I'd prefer that was raised and discussed in the DM list or [CoordinateDM issues](https://github.com/ivoa-std/CoordinateDM) before going back for review.  I've already committed the changes to an rfc2_updates branch.

**Comment by lmichel** [comment]
Actually I was  rather suggesting 
 
```
{ ra :real,
  dec: real,
  unit:string
}
```
But this is **not the topic**.

For a 3D spaces, having different units may make sense (e.g. R,Theta, Phi)
My plan was not  bother you with this but just to show a case where not using Quantity is relevant.
You might raise the question on the list for sure.
